### PR TITLE
Change instances of 'charge' to 'allocate'

### LINKED ISSUE #609: Wording changes: "charged" -> "allocated"
Where we refer to "charging credits", this needs to be changed to "allocating credits" (or whatever makes sense in the context of the sentence structure).

We've already charged for the credits, and they're being allocated, not charged.

### PR DESCRIPTION
**Pull request by DavidMarchant**
As with #632 I've done my best to be exhaustive but if there is any generation of user-facing text outside of app/ I may need a heads up to to alter that too
Again I've erred on the side of over-eager with my changes so if you see any that you'd prefer not in there I can change them back
Closes #609

### CODE REVIEW

File: spec/models/case_spec.rb [hunk 1]
```diff
@@ -436,15 +436,15 @@
   end
 
   describe '#credit_charge' do
-    it 'cannot have a charge less than that of an attached completed change request' do
+    it 'cannot have a  less than that of an attached completed change request' do
```
Comment (jamesremuscat) [inline spec/models/case_spec.rb]: There's a missing word here.

File: app/decorators/credit_charge_decorator.rb [hunk 2]
```diff
@@ -13,13 +13,14 @@ def credit_usage_card
   end
 
   def event_card
+    conjunction = amount == 1 ? 'was' : 'were'
     h.render 'cases/event',
              admin_only: false,
              date: effective_date,
              name: user.name,
-             text:  "A charge of #{h.pluralize(amount, 'credit')} was added for this case.",
+             text:  "#{h.pluralize(amount, 'credit')} #{conjunction}  allocated for this case.",
```
Comment (jamesremuscat) [inline app/decorators/credit_charge_decorator.rb]: There's a double-space between `#{conjunction}` and `allocated`.

File: app/views/change_requests/show.html.erb [hunk 3]
```diff
@@ -15,7 +15,7 @@
                 <td><%= cr.user_facing_state %></td>
               </tr>
               <tr>
-                <th>Credit charge</th>
+                <th>Credit Allocation</th>
```
Comment (jamesremuscat) [inline app/views/change_requests/show.html.erb]: Please can this remain sentence-case, to match how we label other fields?

File: app/views/change_requests/state_controls/_awaiting_authorisation.html.erb [hunk 4]
```diff
@@ -29,8 +29,8 @@
               class: 'btn btn-danger form-control',
               data: {
                   confirm: 'Are you sure you want to decline this change
-                            request? You may still be charged for work carried
-                            out up to this point.'.squish
+                            request? You may still have credits allocated
```
[the remark this answers is not in the document]
Comment (DavidMarchant) [reply app/views/change_requests/state_controls/_awaiting_authorisation.html.erb]: Ah yeah, sure

File: spec/models/case_spec.rb (hunk 1 above)
Comment (DavidMarchant) [reply spec/models/case_spec.rb]: Ah, right you are, sorry

File: app/decorators/credit_charge_decorator.rb (hunk 2 above)
Comment (DavidMarchant) [reply app/decorators/credit_charge_decorator.rb]: Cheers, apologies, changed

File: app/views/change_requests/show.html.erb (hunk 3 above)
Comment (DavidMarchant) [reply app/views/change_requests/show.html.erb]: Yep sure, sorry

**Pull request merged**